### PR TITLE
Added type hints

### PR DESCRIPTION
**Pull request by NatKarmios**

### CODE REVIEW

**Comment by NatKarmios** [comment]
This should hopefully fix all the dupe commits.

File: src/utility.py [hunk 1]
```diff
@@ -19,7 +21,8 @@ def get_channel_by_name(server, name):
 
 
 @commands.has_role('Moderator')
-async def send(bot, message, channel, delete=False, time=MESSAGE_SLEEP_TIME):
+async def send(bot: commands.Bot, message: str, channel: Channel, delete: bool=False,
```
Comment (tom-sherman) [inline src/utility.py]: Remove `delete` hint

File: src/utility.py [hunk 2]
```diff
@@ -19,7 +21,8 @@ def get_channel_by_name(server, name):
...
+               time: float=MESSAGE_SLEEP_TIME) -> None:
```
Comment (tom-sherman) [inline src/utility.py]: Remove `time` hint

File: src/globals.py [hunk 3]
```diff
@@ -2,17 +2,19 @@
...
+from typing import Set
+
+conn: sqlite3.Connection = sqlite3.connect('aryas.db')
 
 with open('secrets.json') as data_file:
     SECRETS = json.load(data_file)
 
 # The possible status an User can have
-possible_status = {'away', 'vacation', 'active'}
+possible_status: Set[str] = {'away', 'vacation', 'active'}
```
Comment (tom-sherman) [inline src/globals.py]: Remove hint

File: src/globals.py [hunk 4]
```diff
@@ -2,17 +2,19 @@
...
 
 
 MESSAGE_SLEEP_TIME = 2
 MOD_LOG_CHANNEL_NAME = 'mod_log'
 
 # Setup logging.
-logger = logging.getLogger('discord')
+logger: logging.Logger = logging.getLogger('discord')
```
Comment (tom-sherman) [inline src/globals.py]: Remove hint

File: src/extensions/modtools.py [hunk 5]
```diff
@@ -1,63 +1,63 @@
...
+            await self.bot.kick(member)
+            msg = '{} was kicked by {}. Reason: {}'.format(member.name, ctx.message.author.mention, reason)
+            await send(self.bot, msg, get_channel_by_name(ctx.message.server, MOD_LOG_CHANNEL_NAME))
+        except Exception as e:
+            print(e)
+            await send(self.bot, 'Failed to kick ' + member.mention, ctx.message.channel, True)
+
+    @commands.command(pass_context=True)
+    @commands.has_role('Admin')
+    async def ban(self, ctx: commands.Context, member: discord.Member, reason: str) -> None:
```
Comment (tom-sherman) [inline src/extensions/modtools.py]: Remove `reason` hint

File: src/extensions/modtools.py [hunk 6]
```diff
@@ -1,63 +1,63 @@
...
+from src.globals import MOD_LOG_CHANNEL_NAME
+
+
+class ModTools:
+    def __init__(self, bot):
+        self.bot: commands.Bot = bot
+
+    @commands.command(pass_context=True)
+    @commands.has_role('Admin')
+    async def kick(self, ctx: commands.Context, member: discord.Member, reason: str) -> None:
```
Comment (tom-sherman) [inline src/extensions/modtools.py]: Remove `reason` hint

File: src/extensions/general.py [hunk 7]
```diff
@@ -1,149 +1,121 @@
...
+                  (giver.id,
+                   member.id,
+                   channel,
+                   server,
+                   love))
+        conn.commit()
+        await send(self.bot, '{} showed {}x❤ to {}'.format(giver.mention, love, member.mention), ctx.message.channel)
+
+    @commands.command(pass_context=True)
+    async def weather(self, ctx: commands.Context, country: str, city: str) -> None:
```
Comment (tom-sherman) [inline src/extensions/general.py]: Remove `city` and `country` hint

File: src/extensions/general.py [hunk 8]
```diff
@@ -1,149 +1,121 @@
...
+from src.utility import send
+import time
+
+
+class General:
+    def __init__(self, bot):
+        self.bot: commands.Bot = bot
+
+    @commands.command(pass_context=True)
+    async def ping(self, ctx) -> None:
```
Comment (tom-sherman) [inline src/extensions/general.py]: Add a hint for `ctx`

**Review by tom-sherman — CHANGES_REQUESTED** [review]
Keep in function param hints where the value could be ambiguous, remove otherwise.

**Pull request merged**